aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -13,51 +12,51 @@ namespace {
 
 using namespace vec256;
 
-template <class scalar_t, class F>
-static void parallel_apply(Tensor& result, const Tensor& self, F f) {
-  internal::init_tbb_num_threads();
-
-static default_partitioner_type ap;
-
-  auto arr_out = result.data<scalar_t>();
-  auto arr_in = self.data<scalar_t>();
-  int64_t size = self.numel();
-  int64_t grain_size = 2048;
-  if (size < grain_size) {
-    map(f, arr_out, arr_in, size);
-  } else {
-    tbb::parallel_for(
-        tbb::blocked_range<int64_t>(0, size, grain_size),
-        [&](const tbb::blocked_range<int64_t>& r) {
-          map(f, arr_out + r.begin(), arr_in + r.begin(), r.end() - r.begin());
-        },
-        ap);
-  }
-}
-
 static void abs_kernel(Tensor& result, const Tensor& self) {
   AT_DISPATCH_ALL_TYPES(self.type(), "abs", [&] {
-    parallel_apply<scalar_t>(
-        result,
-        self,
-        [](const Vec256<scalar_t>& x) { return x.abs(); });  });
+    scalar_t* out = result.data<scalar_t>();
+    scalar_t* in = self.data<scalar_t>();
+    int64_t size = self.numel();
+    parallel_for(0, size, 2048, [out, in](int64_t begin, int64_t end) {
+      map([](const Vec256<scalar_t>& x) { return x.abs(); },
```

Comment:
Where's `map` coming from? BTW it might be possible to use `&Vec<scalar_t>::abs` as the first arg.

Reply:
It's from vec256 functional. I'll try what you suggested.